lib/Sema/CSGen.cpp

```diff
@@ -1345,8 +1345,10 @@ namespace {
       // We may, alternatively, want to use a type variable in that case,
       // and possibly infer the type of the variable that way.
       CS.getTypeChecker().validateDecl(E->getDecl());
-      if (E->getDecl()->isInvalid())
+      if (E->getDecl()->isInvalid()) {
+        E->setType(E->getDecl()->getInterfaceType());
         return nullptr;
+      }
```

Comment:
Right, I've actually used that a couple times—sorry! Would the correct line be this?

    CS.setType(E, E->getDecl()->getInterfaceType());

You mention `ConstraintSystem::getType()`, but I don't think that works on `Decl`s.